libgnucash/engine/gnc-date.cpp

```diff
@@ -585,6 +607,8 @@ void qof_date_completion_set(QofDateCompletion dc, int backmonths)
 */
 const gchar *qof_date_format_get_string(QofDateFormat df)
```

Comment:
This affects date string output as well as parsing. That breaks localized dates on checks, so one gets e.g. 18/05/17 instead of ۸۱/۵۰/۷۱ (which is actually backwards because this editor doesn't support RTL). It probably also breaks fancy dates on invoices, but I can't read Farsi well enough to get that set up. Rather than munging qof_date_format_get_string and qof_date_text_format_get_string you should apply normalize_dt_format only for passing to strptime and to the one strftime call in this file where the string is used to determine the date separator.